DemoInfo/DemoParser.cs

```diff
@@ -506,9 +527,142 @@ public void ParseHeader()
 
 			Header = header;
 
-
 			if (HeaderParsed != null)
 				HeaderParsed(this, new HeaderParsedEventArgs(Header));
+
+				if (header.PlaybackTime == 0)  //header is corrupted
+					FixHeader();
+		}
+
+		void FixHeader()
+		{
+			ExpressParseToEnd();
+			Header.PlaybackTicks = IngameTick;
+			Header.PlaybackFrames = CurrentTick - (TickAtStart + 1);
+			Header.PlaybackTime = TickInterval * IngameTick;
+			
+			CurrentTick = 0;
+
+			//the ol' switcheroo
+			//It's probably safest and easiest to
+			//swap out for a fresh bitstream.
+			Input.Seek(0, System.IO.SeekOrigin.Begin);
+			BitStream = BitStreamUtil.Create(Input);
+			//Put the new stream back at the end of the header
+			ExpressParseHeader(BitStream);
+		}
+
+		//These express functions mirror functions without the express prefix,
+		//but all they do is read ticks and get ServerInfo.  They do not parse
+		//data from any other packets.
```

Comment:
I started with that, but it got kind of messy.  Each call to a parser in ParseTick and ParseDemoPacket needs to be conditional on whether we're skipping the parsing or doing a normal run.  It also seemed cleaner to isolate the logic because 99% of the time when someone's looking at the code they'll be looking at it for normal parsing purposes and the additional logic will just be in the way.

If you'd prefer it the other way though, I have no problem implementing it like that.